core/src/trezor/ui/layouts/tt.py

```diff
@@ -521,64 +523,153 @@ async def confirm_decred_sstx_submission(
     )
 
 
-async def confirm_hex(
+async def confirm_blob(
     ctx: wire.GenericContext,
     br_type: str,
     title: str,
-    data: str,
-    subtitle: str | None = None,
+    data: bytes | str,
     description: str | None = None,
     br_code: ButtonRequestType = ButtonRequestType.Other,
     icon: str = ui.ICON_SEND,  # TODO cleanup @ redesign
     icon_color: int = ui.GREEN,  # TODO cleanup @ redesign
-    color_description: int = ui.FG,  # TODO cleanup @ redesign
-    width: int | None = MONO_HEX_PER_LINE,
-    width_paginated: int = MONO_HEX_PER_LINE - 2,
-    truncate: bool = False,
-    truncate_middle: bool = False,
-    truncate_ellipsis: str = "...",
 ) -> None:
-    if truncate:
+    """Confirm data blob.
+
+    Applicable for public keys, signatures, hashes. In general, any kind of
+    data that is not human-readable, and can be wrapped at any character.
+
+    For addresses, use `confirm_address`.
+
+    Displays in monospace font. Paginates automatically.
+    If data is provided as bytes or bytearray, it is converted to hex.
+    """
+    if isinstance(data, (bytes, bytearray)):
+        data_str = hexlify(data).decode()
+    else:
+        data_str = data
+
+    span = Span()
+    lines = 0
+    if description is not None:
+        span.reset(description, 0, ui.NORMAL)
+        lines += span.count_lines()
+    lines += (len(data_str) + MONO_HEX_PER_LINE - 1) // MONO_HEX_PER_LINE
+
+    if lines <= TEXT_MAX_LINES:
         text = Text(title, icon, icon_color, new_lines=False)
-        description_lines = 0
-        if subtitle is not None:
-            description_lines += Span(subtitle, 0, ui.BOLD).count_lines()
-            text.bold(subtitle)
-            text.br()
         if description is not None:
-            description_lines += Span(description, 0, ui.NORMAL).count_lines()
-            text.content.extend((ui.NORMAL, color_description, description, ui.FG))
+            text.normal(description)
             text.br()
-        if width is not None:
-            text.mono(
-                *_truncate_hex(
-                    data,
-                    lines=TEXT_MAX_LINES - description_lines,
-                    width=width,
-                    middle=truncate_middle,
-                    ellipsis=truncate_ellipsis,
-                )
-            )
+        # special case:
```

Comment:
it can't beacause it only applies when len is divisible by 16
that means that it must be 64 or less, the next allowable value up is 80 which is 5 lines of 16 chars, and also 5 lines of 18 chars

but good catch anyway, the assert should be changed to something like `(len // MONO_HEX_PER_LINE >= len // 16)`

Reply:
explained in b1b284b